qcodes/instrument_drivers/rohde_schwarz/ZNB.py

```diff
@@ -346,25 +497,42 @@ def _set_stop(self, val: float):
         if val != stop:
             log.warning(
                 f"Could not set stop to {val} setting it to {stop}")
-        self.update_traces()
+        self.update_lin_traces()
 
     def _set_npts(self, val: int):
         channel = self._instrument_channel
         self.write(f'SENS{channel}:SWE:POIN {val:.7f}')
-        self.update_traces()
+        self.update_lin_traces()
+        self.update_cw_traces()
+
+    def _set_bandwidth(self, val: int):
+        channel = self._instrument_channel
+        self.write(f'SENS{channel}:BAND {val:.4f}')
+        self.update_cw_traces()
 
     def _set_span(self, val: float):
         channel = self._instrument_channel
         self.write(f'SENS{channel}:FREQ:SPAN {val:.7f}')
-        self.update_traces()
+        self.update_lin_traces()
 
     def _set_center(self, val: float):
         channel = self._instrument_channel
         self.write(f'SENS{channel}:FREQ:CENT {val:.7f}')
-        self.update_traces()
+        self.update_lin_traces()
+
+    def _set_sweep_type(self, val: str):
+        channel = self._instrument_channel
+        self.write(f'SENS{channel}:SWE:TYPE {val}')
+
+    def _set_cw_frequency(self, val: float):
+        channel = self._instrument_channel
+        self.write(f'SENS{channel}:FREQ:CW {val:.7f}')
 
-    def update_traces(self):
-        """ updates start, stop and npts of all trace parameters"""
+    def update_lin_traces(self):
```

Comment:
renaming this is an api break. We can add back `update_traces` as a deprecated method that simply calls `update_lin_traces`